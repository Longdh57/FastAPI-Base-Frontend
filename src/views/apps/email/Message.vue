<template>
  <div>
    <MailToolbar/>

    <div class="c-message p-3">
      <div class="c-message-details">
        <div class="c-message-headers">
          <div class="c-message-headers-subject">Lorem ipsum dolor sit amet</div>
          <div class="c-message-headers-from">
            Lukasz Holeczek
            <span class="text-muted">email@email.com</span>
          </div>
          <div class="c-message-headers-date">Today, <b>3:47 PM</b></div>
        </div>
        <hr>
        <div class="c-message-body">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
          <blockquote>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </blockquote>
        </div>
        <div class="c-message-attachments">
          <div class="c-message-attachment">
            <CBadge color="danger" class="mr-1">zip</CBadge> 
            <b>bootstrap.zip</b> <i>(2,5MB)</i>
            <span class="ml-auto">
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-magnifying-glass"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-share"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-cloud-download"/>
              </CButton>
            </span>
          </div>
          <div class="c-message-attachment">
            <CBadge color="info" class="mr-1">txt</CBadge> 
            <b>readme.txt</b> <i>(7KB)</i>
            <span class="ml-auto">
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-magnifying-glass"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-share"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-cloud-download"/>
              </CButton>
            </span>
          </div>
          <div class="c-message-attachment">
            <CBadge color="success" class="mr-1">xls</CBadge> 
            <b>spreadsheet.xls</b> <i> (984KB)</i>
            <span class="ml-auto">
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-magnifying-glass"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-share"/>
              </CButton>
              <CButton small class="btn-link" href="#">
                <CIcon name="cil-cloud-download"/>
              </CButton>
            </span>
          </div>
        </div>
        <form class="mt-3" method="post" action="">
          <div class="form-group">
            <textarea 
              class="form-control" 
              id="message" 
              name="body" 
              rows="12" 
              placeholder="Click here to reply"
            ></textarea>
          </div>
          <div class="form-group">
            <CButton color="success" tabindex="3" type="submit">
              Send message
            </CButton>
          </div>
        </form>
      </div>
    </div>
  </div>
</template>

<script>
import MailToolbar from './MailToolbar'

export default {
  name: 'Message',
  components: {
    MailToolbar
  }
}
</script>
