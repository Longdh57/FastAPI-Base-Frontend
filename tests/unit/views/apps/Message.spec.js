import {shallowMount, createLocalVue} from '@vue/test-utils';
import VueRouter from 'vue-router'
import CoreuiVue from '@coreui/vue-pro'
import Message from '@/views/apps/email/Message'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('Message.vue', () => {
  it('has a name', () => {
    expect(Message.name).toMatch('Message')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Message, {
      localVue,
      router
    })
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Message', () => {
    const wrapper = shallowMount(Message, {
      localVue,
      router
    })
    expect(wrapper.findComponent(Message)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Message, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
