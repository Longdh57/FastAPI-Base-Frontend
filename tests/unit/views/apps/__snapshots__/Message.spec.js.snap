// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.vue renders correctly 1`] = `
<div>
  <mailtoolbar-stub />
   
  <div
    class="c-message p-3"
  >
    <div
      class="c-message-details"
    >
      <div
        class="c-message-headers"
      >
        <div
          class="c-message-headers-subject"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <div
          class="c-message-headers-from"
        >
          
          Lukasz Holeczek
          
          <span
            class="text-muted"
          >
            email@email.com
          </span>
        </div>
         
        <div
          class="c-message-headers-date"
        >
          Today, 
          <b>
            3:47 PM
          </b>
        </div>
      </div>
       
      <hr />
       
      <div
        class="c-message-body"
      >
        <p>
          
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        
        </p>
         
        <blockquote>
          
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        
        </blockquote>
      </div>
       
      <div
        class="c-message-attachments"
      >
        <div
          class="c-message-attachment"
        >
          <cbadge-stub
            activeclass="router-link-active"
            class="mr-1"
            color="danger"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            tag="span"
            target="_self"
          >
            zip
          </cbadge-stub>
           
          <b>
            bootstrap.zip
          </b>
           
          <i>
            (2,5MB)
          </i>
           
          <span
            class="ml-auto"
          >
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-magnifying-glass"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-share"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-cloud-download"
              />
            </cbutton-stub>
          </span>
        </div>
         
        <div
          class="c-message-attachment"
        >
          <cbadge-stub
            activeclass="router-link-active"
            class="mr-1"
            color="info"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            tag="span"
            target="_self"
          >
            txt
          </cbadge-stub>
           
          <b>
            readme.txt
          </b>
           
          <i>
            (7KB)
          </i>
           
          <span
            class="ml-auto"
          >
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-magnifying-glass"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-share"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-cloud-download"
              />
            </cbutton-stub>
          </span>
        </div>
         
        <div
          class="c-message-attachment"
        >
          <cbadge-stub
            activeclass="router-link-active"
            class="mr-1"
            color="success"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            tag="span"
            target="_self"
          >
            xls
          </cbadge-stub>
           
          <b>
            spreadsheet.xls
          </b>
           
          <i>
             (984KB)
          </i>
           
          <span
            class="ml-auto"
          >
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-magnifying-glass"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-share"
              />
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="router-link-active"
              class="btn-link"
              event="click"
              exactactiveclass="router-link-exact-active"
              href="#"
              routertag="a"
              small=""
              target="_self"
              type="button"
            >
              <cicon-stub
                name="cil-cloud-download"
              />
            </cbutton-stub>
          </span>
        </div>
      </div>
       
      <form
        action=""
        class="mt-3"
        method="post"
      >
        <div
          class="form-group"
        >
          <textarea
            class="form-control"
            id="message"
            name="body"
            placeholder="Click here to reply"
            rows="12"
          />
        </div>
         
        <div
          class="form-group"
        >
          <cbutton-stub
            activeclass="router-link-active"
            color="success"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            tabindex="3"
            target="_self"
            type="submit"
          >
            
            Send message
          
          </cbutton-stub>
        </div>
      </form>
    </div>
  </div>
</div>
`;
